### PR TITLE
Bluetooth: Mesh: Fix time client model

### PR DESCRIPTION
**Pull request by RAlexeev**
Description of functions contradicted their logic. Although functions' declarations allowed response argument to be NULL to keep from blocking, their definitions returned error -EFAULT in this case. And Time client instance didn't have handlers for Status messages.

### CODE REVIEW

**Review by trond-snekvik — COMMENTED** [review]
Thanks for contributing!

The changes look good, but could you please squash the commits into one?

**Comment by RAlexeev** [comment]
Yes, certainly. I hope, now it looks better.

**Review by joerchan — CHANGES_REQUESTED** [review]
Thank you for the contribution.

Changes looks good, but the rules for the pointers should be documented so that users don't have to look into how they are used in the implementation.

@trond-snekvik Looks like other places could do with a documentation update here.
For example gen_lock_srv allows handlers struct and individual handlers to be NULL, but gen_lvl_srv requires both handlers and indiviual handlers to be non-NULL.

File: include/bluetooth/mesh/time_cli.h [hunk 1]
```diff
@@ -57,6 +101,8 @@ struct bt_mesh_time_cli {
 	struct bt_mesh_model_pub pub;
 	/** Acknowledged message tracking. */
 	struct bt_mesh_model_ack_ctx ack_ctx;
+	/** Handler function structure. */
```
[the remark this answers is not in the document]
Comment (RAlexeev) [reply include/bluetooth/mesh/time_cli.h]: Thanks for your comments! I've tried to fix the code according to them. Is everything okay now?

File: include/bluetooth/mesh/time_cli.h [hunk 2]
```diff
@@ -57,6 +129,13 @@ struct bt_mesh_time_cli {
 	struct bt_mesh_model_pub pub;
 	/** Acknowledged message tracking. */
 	struct bt_mesh_model_ack_ctx ack_ctx;
+	/** Collection of handler callbacks.
+	*
+	* @note Function pointers to handlers must point to valid memories,
+	* which have execute permissions.
+	*
+	*/
```
Comment (trond-snekvik) [inline include/bluetooth/mesh/time_cli.h:137]:
```suggestion
	/** Collection of handler callbacks.
	 *
	 * @note Function pointers to handlers must point to valid, executable 
	 * memory.
	 */
```

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
Could you please convert your second commit to follow the commit format described in the failing Github action and rebase on master? I only have one minor comment on the change, but we need to pass CI to get this in.

File: include/bluetooth/mesh/time_cli.h (hunk 2 above)
Comment (joerchan) [reply include/bluetooth/mesh/time_cli.h:137]: I think the documentation here is somewhat misleading. 
The thing we wish to document here is that the structure is not copied right and that storage of this struct is up to the application, right?
This storage does not have to be in executable memory.

I think "Must point to memory that remains valid." is good enough.

Placing the function that the function pointer points to is really up to the compiler, so not something that the user of the API should have to care about.

File: include/bluetooth/mesh/time_cli.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (joerchan) [reply include/bluetooth/mesh/time_cli.h]: Yes, this looks good. Thank you.

**Review by joerchan — APPROVED** [review]
Approving as my requested changes have been addressed.
Please address the changes requested by Trond as well.

File: include/bluetooth/mesh/time_cli.h (hunk 2 above)
Comment (RAlexeev) [reply include/bluetooth/mesh/time_cli.h]: Last time I misunderstood the idea. Yes, indeed, I agree with you. And I have fixed it.

**Comment by RAlexeev** [comment]
> Could you please convert your second commit to follow the commit format described in the failing Github action and rebase on master? I only have one minor comment on the change, but we need to pass CI to get this in.

Yes, sorry for a bad commit format, now I fixed it.

**Review by Andrewpini — APPROVED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Pull request merged**